app.js

```diff
@@ -10,110 +17,241 @@ var product1Image = document.getElementById('img1');
+  new Product('img/wine-glass.jpg','Wine Glass')
 
 ];
-//Here I'm saying that these variables are these pictures.
+
 var product1 = allProducts[0];
 var product2 = allProducts[1];
 var product3 = allProducts[2];
 
-//I want to know how many votes does each button get WHEN CLICKED!!!
-//I want a function that tracks that.
-//Where am i listening: I'm listening at buttonNumber1
+Product.uniqueDouble = function () {
+  var uniqueOptions = [];
+  //math.random to generate random number
+  //Account fot the array lenght (allProducts)
+  while(uniqueOptions.length < 3){
+    var ranNum = Math.floor(Math.random() * allProducts.length);
+    //compare 2 previous with the new one
+    //Includes gives me a boolean
+    //!Whatever this evaluates, change it
+    //compare the 2 numbers to each other
+    //&& bot of those to be true
+    if(!Product.lastImgage.includes(ranNum) && !uniqueOptions.includes(ranNum)) {
+      uniqueOptions.push(ranNum);
+      console.log(uniqueOptions);
+    } else {
+      console.log('Duplicate!');
+      console.log(uniqueOptions);
+    }
+  }
+  Product.lastImgage = uniqueOptions;
+  console.log(uniqueOptions);
+  return uniqueOptions;
+};
+
+
+
+Product.handleClick = function(e) {
+
+  Product.totalClicks++;
+
+  console.log(event.target.alt);
+
+  for(var i in Product.allProducts) {
+    if(e.target.alt === Product.allProducts[i].name) {
+      Product.allProducts[i].votes++;
+      break;
+    }
+  }
+};
+Product.updateVotes = function() {
+  for(var i in Product.allProducts){
+    Product.productNames[i] = Product.allProducts[i].name;
+    Product.productVotes[i] = Product.allProducts[i].votes;
+
+    Product.arrayOfColors.push('#' + Math.floor(Math.random() * 16777215).toString(16));
+
+  }
+};
+
+Product.renderChart = function () {
+  var ctx = document.getElementById('myChart');
 
-buttonNumber1.addEventListener('click', function (e){
-  //I WANT THIS FUNCTION TO INCREMENT THE VOTE COUNT of this button
+  new Chart(ctx, {
+    type: 'bar',
+    data: {
+      labels: Product.productNames,
+      datasets: [{
+        label:'Votes per product',
+        data: Product.productVotes,
+        backgroundColor: Product.arrayOfColors,
+        hoverBackgroundColor: 'yellow'
+      }]
+    },
+    options: {
+      scales:{
+        yAxes: [{
+          ticks: {
+            beginAtZero: true
+          }
+        }]
+      },
+      title: {
+        display: true,
+        text:'Results'
+      }
+    }
+  });
+};
+console.log('chart');
+
+//percentage
+
+//Here I'm saying that these variables are these pictures.
+
+
+var ulElement = document.getElementById('counter-list');
+
+function renderList(){
+  for(var i = 0; i < allProducts.length; i++){
+    var listElement = document.createElement('li');
+    listElement.textContent = allProducts[i].name + ' ' + allProducts[i].votes;
+    ulElement.appendChild(listElement);
+  }
+}
+
+function handleClickImg1() {
+  counter++;
   product1.votes++;
+  stopImages();
```

Comment:
Since you increment `counter` and call `stopImages` in all three of these click handlers, you could move that functionality into the `changeImage` function and only have to write those two lines once.